### PR TITLE
fix: Add Retrieval SK Quad dataset for Slovak search evaluation

### PR DESCRIPTION
**Pull request by Kroli99**
This commit introduces the Retrieval SK Quad dataset, designed to assess Slovak search performance. The dataset is derived from SK-QuAD and includes questions with their best answers categorized post-annotation. This addition provides a significant resource for advancing Slovak language search evaluation and supporting further research and development.

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding datasets checklist

**Reason for dataset addition**: ... 

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/tasks/Retrieval/slk/SKQuadRetrieval.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        dataset={
+            "path": "TUKE-KEMT/retrieval-skquad",
+            "revision": "09f81f51dd5b8497da16d02c69c98d5cb5993ef2",
+        },
+        type="Retrieval",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["slk-Latn"],
+        main_score="ndcg_at_10",
+        date=None,
```
Comment (isaac-chung) [inline mteb/tasks/Retrieval/slk/SKQuadRetrieval.py:28]: Please add this.

File: mteb/tasks/Retrieval/slk/SKQuadRetrieval.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from datasets import load_dataset
+
+from mteb.abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SKQuadRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
```
Comment (isaac-chung) [inline mteb/tasks/Retrieval/slk/SKQuadRetrieval.py:10]: Please add citation.

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Thanks for opening this. A few small things.

Also, were you able to run `mteb run -m {model_name} -t {task_name}` for a model? @KennethEnevoldsen should we commit the result to the [results repo](https://github.com/embeddings-benchmark/results)?

**Comment by KennethEnevoldsen** [comment]
> @KennethEnevoldsen should we commit the result to the [results repo](https://github.com/embeddings-benchmark/results)?

Would be great to have a way to inspect the results at least. Might be worth making a CLI command for pushing runs to the results repo, WDYT?

**Comment by isaac-chung** [comment]
> Would be great to have a way to inspect the results at least. Might be worth making a CLI command for pushing runs to the results repo, WDYT?

Adding this as a separate issue. 

@Kroli99 feel free to attach the results here to open a PR at the results repo with the results from this PR.

**Comment by Kroli99** [comment]
I added all the requested changes. 
If i add from .slk.SKQuadRetrieval import * to the init.py file I receive 1 failed test. Without everything is alright. 
Can you give me some insights on this?

File: mteb/tasks/Retrieval/slk/SKQuadRetrieval.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            }
+            self.relevant_docs = {"test": {}}
+
+            for row in ds_default["test"]:
+                self.relevant_docs["test"].setdefault(row["query-id"], {})[
+                    row["corpus-id"]
+                ] = int(row["score"])
+
+            print(
+                f"Data Loaded:\n- Corpus size: {len(self.corpus['test'])}\n- Query size: {len(self.queries['test'])}\n- Relevance entries: {len(self.relevant_docs['test'])}"
```
Comment (Samoed) [inline mteb/tasks/Retrieval/slk/SKQuadRetrieval.py:77]: I think that you should remove print

File: mteb/tasks/Retrieval/slk/SKQuadRetrieval.py (hunk 2 above)
Comment (isaac-chung) [reply mteb/tasks/Retrieval/slk/SKQuadRetrieval.py:10]: Is there no citation for this dataset? I'm not sure if that aligns with our goals if there isn't any. cc @KennethEnevoldsen

**Comment by isaac-chung** [comment]
Results look fine.

<details>
<summary>intfloat/multilingual-e5-small</summary>

```
{
  "dataset_revision": "09f81f51dd5b8497da16d02c69c98d5cb5993ef2",
  "evaluation_time": 15.720594644546509,
  "kg_co2_emissions": null,
  "mteb_version": "1.14.21",
  "scores": {
    "test": [
      {
        "hf_subset": "default",
        "languages": [
          "slk-Latn"
        ],
        "main_score": 0.68128,
        "map_at_1": 0.40828,
        "map_at_10": 0.5559,
        "map_at_100": 0.57091,
        "map_at_1000": 0.57164,
        "map_at_20": 0.56514,
        "map_at_3": 0.52041,
        "map_at_5": 0.53972,
        "mrr_at_1": 0.6067019400352733,
        "mrr_at_10": 0.7488788107835738,
        "mrr_at_100": 0.7504650245926571,
        "mrr_at_1000": 0.7505040587200038,
        "mrr_at_20": 0.750175768347117,
        "mrr_at_3": 0.7330981775426231,
        "mrr_at_5": 0.7447824808935933,
        "nauc_map_at_1000_diff1": 0.2866992517620442,
        "nauc_map_at_1000_max": 0.10431750981757615,
        "nauc_map_at_1000_std": -0.08487662998940329,
        "nauc_map_at_100_diff1": 0.2864088375784633,
        "nauc_map_at_100_max": 0.10434889094799742,
        "nauc_map_at_100_std": -0.0843278610086469,
        "nauc_map_at_10_diff1": 0.28363754881883757,
        "nauc_map_at_10_max": 0.09666885000686007,
        "nauc_map_at_10_std": -0.0980052595631516,
        "nauc_map_at_1_diff1": 0.33320502101180705,
        "nauc_map_at_1_max": 0.06894286817651117,
        "nauc_map_at_1_std": -0.10079744178764052,
        "nauc_map_at_20_diff1": 0.2860967940986825,
        "nauc_map_at_20_max": 0.10148615146993561,
        "nauc_map_at_20_std": -0.08896708103697706,
        "nauc_map_at_3_diff1": 0.27576736519966666,
        "nauc_map_at_3_max": 0.07976614662883112,
        "nauc_map_at_3_std": -0.12002779320933135,
        "nauc_map_at_5_diff1": 0.28179090487002295,
        "nauc_map_at_5_max": 0.08813648090757556,
        "nauc_map_at_5_std": -0.10816457085248135,
        "nauc_mrr_at_1000_diff1": 0.37112026477180565,
        "nauc_mrr_at_1000_max": 0.2235192858858728,
        "nauc_mrr_at_1000_std": -0.04474250361863392,
        "nauc_mrr_at_100_diff1": 0.371084992589638,
        "nauc_mrr_at_100_max": 0.22363789527148797,
        "nauc_mrr_at_100_std": -0.044554257911297704,
        "nauc_mrr_at_10_diff1": 0.3713578547546789,
        "nauc_mrr_at_10_max": 0.22736911895988282,
        "nauc_mrr_at_10_std": -0.042718485137591966,
        "nauc_mrr_at_1_diff1": 0.39067535767332157,
        "nauc_mrr_at_1_max": 0.14974889335127395,
        "nauc_mrr_at_1_std": -0.06514514596943974,
        "nauc_mrr_at_20_diff1": 0.3711047947652698,
        "nauc_mrr_at_20_max": 0.22432896795736548,
        "nauc_mrr_at_20_std": -0.04427006814103498,
        "nauc_mrr_at_3_diff1": 0.3608248871000455,
        "nauc_mrr_at_3_max": 0.22809405344141323,
        "nauc_mrr_at_3_std": -0.05875523165825909,
        "nauc_mrr_at_5_diff1": 0.3684949644288355,
        "nauc_mrr_at_5_max": 0.22843802921938636,
        "nauc_mrr_at_5_std": -0.043570171089939616,
        "nauc_ndcg_at_1000_diff1": 0.31802034788721273,
        "nauc_ndcg_at_1000_max": 0.17122001257265773,
        "nauc_ndcg_at_1000_std": -0.035054268640823026,
        "nauc_ndcg_at_100_diff1": 0.31255844597409277,
        "nauc_ndcg_at_100_max": 0.17234567544099946,
        "nauc_ndcg_at_100_std": -0.02186075185974356,
        "nauc_ndcg_at_10_diff1": 0.3051586058655539,
        "nauc_ndcg_at_10_max": 0.15711286264360935,
        "nauc_ndcg_at_10_std": -0.05741543196762416,
        "nauc_ndcg_at_1_diff1": 0.39641254068035264,
        "nauc_ndcg_at_1_max": 0.14489961449338437,
        "nauc_ndcg_at_1_std": -0.0628991953200473,
        "nauc_ndcg_at_20_diff1": 0.31045482332727425,
        "nauc_ndcg_at_20_max": 0.15990125048256515,
        "nauc_ndcg_at_20_std": -0.04070505779780054,
        "nauc_ndcg_at_3_diff1": 0.2903109869548275,
        "nauc_ndcg_at_3_max": 0.14534346170953527,
        "nauc_ndcg_at_3_std": -0.09919888662798655,
        "nauc_ndcg_at_5_diff1": 0.29968511340831594,
        "nauc_ndcg_at_5_max": 0.14684481051512743,
        "nauc_ndcg_at_5_std": -0.07560669357020834,
        "nauc_precision_at_1000_diff1": -0.09319321468900274,
        "nauc_precision_at_1000_max": 0.13490733960255447,
        "nauc_precision_at_1000_std": 0.16689865489002267,
        "nauc_precision_at_100_diff1": -0.07586759634209606,
        "nauc_precision_at_100_max": 0.1728274581077412,
        "nauc_precision_at_100_std": 0.23529850420404352,
        "nauc_precision_at_10_diff1": -0.02203884403323463,
        "nauc_precision_at_10_max": 0.19826522316860296,
        "nauc_precision_at_10_std": 0.14624897416613886,
        "nauc_precision_at_1_diff1": 0.39067535767332157,
        "nauc_precision_at_1_max": 0.14974889335127395,
        "nauc_precision_at_1_std": -0.06514514596943974,
        "nauc_precision_at_20_diff1": -0.03961846134981961,
        "nauc_precision_at_20_max": 0.17261219317969798,
        "nauc_precision_at_20_std": 0.1954359002963541,
        "nauc_precision_at_3_diff1": 0.06003936956890915,
        "nauc_precision_at_3_max": 0.22452310330657455,
        "nauc_precision_at_3_std": -0.019822287664856695,
        "nauc_precision_at_5_diff1": 0.016295978299353277,
        "nauc_precision_at_5_max": 0.22496838364792204,
        "nauc_precision_at_5_std": 0.08065504683251118,
        "nauc_recall_at_1000_diff1": 0.14577481269199835,
        "nauc_recall_at_1000_max": 0.37067748622952795,
        "nauc_recall_at_1000_std": 0.614585714422309,
        "nauc_recall_at_100_diff1": 0.10966306993151789,
        "nauc_recall_at_100_max": 0.20282791811417406,
        "nauc_recall_at_100_std": 0.368670727895303,
        "nauc_recall_at_10_diff1": 0.1574620970271047,
        "nauc_recall_at_10_max": 0.08189359371697832,
        "nauc_recall_at_10_std": -0.0076345919453605785,
        "nauc_recall_at_1_diff1": 0.33320502101180705,
        "nauc_recall_at_1_max": 0.06894286817651117,
        "nauc_recall_at_1_std": -0.10079744178764052,
        "nauc_recall_at_20_diff1": 0.1512012274298296,
        "nauc_recall_at_20_max": 0.08040350485064324,
        "nauc_recall_at_20_std": 0.07363310558190785,
        "nauc_recall_at_3_diff1": 0.17516781427766334,
        "nauc_recall_at_3_max": 0.06298442161896568,
        "nauc_recall_at_3_std": -0.11827084487361213,
        "nauc_recall_at_5_diff1": 0.17740213593788434,
        "nauc_recall_at_5_max": 0.06459531333008349,
        "nauc_recall_at_5_std": -0.06770245927428464,
        "ndcg_at_1": 0.59735,
        "ndcg_at_10": 0.68128,
        "ndcg_at_100": 0.71724,
        "ndcg_at_1000": 0.72653,
        "ndcg_at_20": 0.69861,
        "ndcg_at_3": 0.63532,
        "ndcg_at_5": 0.65878,
        "precision_at_1": 0.6067,
        "precision_at_10": 0.12407,
        "precision_at_100": 0.01635,
        "precision_at_1000": 0.0018,
        "precision_at_20": 0.06997,
        "precision_at_3": 0.32393,
        "precision_at_5": 0.21799,
        "recall_at_1": 0.40828,
        "recall_at_10": 0.74681,
        "recall_at_100": 0.90649,
        "recall_at_1000": 0.97809,
        "recall_at_20": 0.81147,
        "recall_at_3": 0.62247,
        "recall_at_5": 0.68197
      }
    ]
  },
  "task_name": "SKQuadRetrieval"
}
```

</details>

**Review by isaac-chung — APPROVED** [review]
Thanks for adding this. Just wanted to get a resolution on citation before merging.

File: mteb/tasks/Retrieval/slk/SKQuadRetrieval.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Retrieval/slk/SKQuadRetrieval.py:10]: Then simply add:

citation=""

Comment (isaac-chung) [reply mteb/tasks/Retrieval/slk/SKQuadRetrieval.py:10]: Will use empty string.

**Pull request merged**